core/src/main/java/org/infinispan/container/EntryFactoryImpl.java

```diff
@@ -209,6 +209,18 @@ private CacheEntry getFromContainer(Object key, boolean isOwner, boolean writeOp
       return null;
    }
 
+   private CacheEntry getFromContainerForRead(Object key, boolean isOwner) {
+      InternalCacheEntry ice = container.get(key);
+      if (trace) {
+         log.tracef("Retrieved from container %s", ice);
+      }
+      if (isOwner) {
+         return ice == null ? NullCacheEntry.getInstance() : ice;
+      } else {
+         return ice == null || !ice.isL1Entry() ? null : ice;
```

Comment:
Right, I missed that we ignore non-L1 entries here. Then the only difference from getFromContainer() is that this method is skipping the if (writeOperation) check in innerGetFromContainer()?